test/unit/amp/infinite-scroll-helper.js

```diff
@@ -0,0 +1,262 @@
+      publisherConfig: dummyPublisherConfig,
+    });
+    const inlineConfig = await infiniteScrollAmp.getInitialInlineConfig({
+      itemsToTake: 5,
+    });
+    assert.strictEqual(inlineConfig instanceof Error, true);
+    assert.throws(() => {
+      throw new Error("Required params for getInitialInlineConfig missing");
+    }, inlineConfig);
+  });
+  it("should return null if infinite scroll collection doesn't exist", async function () {
+    const clientStub = getClientStub({
+      getCollectionBySlug: (slug) =>
+        new Promise((resolve) => {
+          resolve(null);
+        }),
+    });
+    const infiniteScrollAmp = new InfiniteScrollAmp({
+      ampConfig: {},
+      client: clientStub,
+      publisherConfig: dummyPublisherConfig,
+    });
+    const inlineConfig = await infiniteScrollAmp.getInitialInlineConfig({
+      itemsToTake: 5,
+      storyId: 2222,
+    });
+    assert.strictEqual(inlineConfig, null);
+  });
+  it("should return null if infinite scroll collection contains no stories", async function () {
+    const clientStub = getClientStub({
+      getCollectionBySlug: (slug) =>
+        new Promise((resolve) => {
+          resolve({
+            items: [],
+          });
+        }),
+    });
+    const infiniteScrollAmp = new InfiniteScrollAmp({
+      ampConfig: {},
+      client: clientStub,
+      publisherConfig: dummyPublisherConfig,
+    });
+    const inlineConfig = await infiniteScrollAmp.getInitialInlineConfig({
+      itemsToTake: 5,
+      storyId: 2222,
+    });
+    assert.strictEqual(inlineConfig, null);
+  });
+  it("should remove current story from infinite scroll", async function () {
+    const clientStub = getClientStub();
+    const infiniteScrollAmp = new InfiniteScrollAmp({
+      ampConfig: {},
+      client: clientStub,
+      publisherConfig: dummyPublisherConfig,
+    });
+    const inlineConfig = await infiniteScrollAmp.getInitialInlineConfig({
+      itemsToTake: 5,
+      storyId: 2222,
+    });
+    assert.strictEqual(false, /sports\/bb/.test(inlineConfig));
+    assert.strictEqual(false, /bb\/b.jpg/.test(inlineConfig));
+  });
+  it("should format JSON as per AMP spec", async function () {
+    // https://amp.dev/documentation/components/amp-next-page/
+    const clientStub = getClientStub();
+    const infiniteScrollAmp = new InfiniteScrollAmp({
+      ampConfig: {},
+      client: clientStub,
+      publisherConfig: dummyPublisherConfig,
+    });
+    const inlineConfig = await infiniteScrollAmp.getInitialInlineConfig({
+      itemsToTake: 5,
+      storyId: 2222,
+    });
+    function isInlineConfigStructureValid(jsonStr) {
+      const stories = JSON.parse(jsonStr);
+      if (!stories.length) throw new Error("Can't verify empty array!");
+      stories.forEach((story) => {
+        const keys = Object.keys(story);
+        if (
+          keys.includes("image") &&
+          keys.includes("url") &&
+          keys.includes("title") &&
+          keys.length === 3
+        )
```

Comment:
wasn't adding `headline` for mock stories before, because of which `title` was not being added. This was incorrect, made the change